lib/index.js

```diff
@@ -96,16 +96,38 @@ function Socket (type, opts) {
             break;
     }
 
-    this.binding = nn.Socket(this.af_domain, this.protocol);
-    this.queue = [];
+  this.binding = nn.Socket(this.af_domain, this.protocol);
+  this.queue = [];
 
-    for(var sokopt in sol){
-        if(opts.hasOwnProperty(sokopt)) this[sokopt](opts[sokopt]);
+  /* subscription filters */
+  this.channels = {};
+
+  /* subscription handling at initialization */
+  if (opts.hasOwnProperty('chan')) {
+    if (typeof opts.chan == 'string') {
+      opts.chan.split(',').forEach(subscriber.bind(this))
+    } else {
```

Comment:
OK but if that thing is neither String nor Array then I want that forEach to throw the process